src/main/kotlin/no/nav/familie/ba/mottak/hendelser/LeesahService.kt

```diff
@@ -175,6 +185,53 @@ class LeesahService(
         oppdaterHendelseslogg(pdlHendelse)
     }
 
+    private fun behandleSivilstandHendelse(pdlHendelse: PdlHendelse) {
+        if (hendelsesloggRepository.existsByHendelseIdAndConsumer(pdlHendelse.hendelseId, CONSUMER_PDL)) {
+            leesahDuplikatCounter.increment()
+            return
+        }
+
+        when (pdlHendelse.endringstype) {
+            OPPRETTET -> {
+                logHendelse(pdlHendelse, "sivilstandDato: ${pdlHendelse.sivilstandDato}")
+                sivilstandOpprettetCounter.increment()
+
+                opprettTaskHvisSivilstandErGift(pdlHendelse)
+            }
+            else -> {
+                logHendelse(pdlHendelse, "Ikke av type OPPRETTET.")
+                when (pdlHendelse.endringstype) {
+                    ANNULLERT -> sivilstandAnnullertCounter.increment()
+                    KORRIGERT -> sivilstandKorrigertCounter.increment().also { opprettTaskHvisSivilstandErGift(pdlHendelse) }
+                }
+                if (pdlHendelse.tidligereHendelseId != null) {
+                    avvikshåndterTidligereTask(pdlHendelse)
```

Comment:
Det som er litt dumt med å gjøre avvikshåndteringer her, er at listeneren har større sjanse for å feile og vi må manuelt fikse.

I tillegg vil det gå tregere hvis man mister offsett og starter fra begynnelsen. 

Er det type på korrigeringer og annuleringer? I så fall bør vi ha det filteret her også.